taqueria-vscode-extension-web-ui/src/App.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useEffect, useState } from 'react';
+import { AllEditors } from './AllEditors';
+import { MichelineEditor } from './data-editors/MichelineEditor';
+import { InteropMessageInterface } from './interopTypes';
+
+const state = {
+	interop: undefined as undefined | InteropMessageInterface,
+	onChangeCallback: (x: InteropMessageInterface) => {/* empty */},
+	logs: [] as { message: string; data?: unknown }[],
+};
+
+// const log = (message: string, data?: unknown) => {
+// 	state.logs.push({ message, data });
+// };
+// console.log = log;
+
+const setGlobalInteropMessageInterface = (x: InteropMessageInterface) => {
+	state.interop = x;
+	state.onChangeCallback(x);
+};
+const w = window as unknown as Record<string, unknown>;
```

Comment:
Why are we casting the window to a Record? I assume we've added a property and using this hack to get around the typesystem.

What would be best is an approach like the following:
```
declare global {
    interface Window { customProp: string; }
}

window.customProp ||= "custom value" 
```

Reply:
The above preserves the Window type, which is useful for code completion and such, important when a developer updates the code to make another reference to the Window object and requires type safety.